rcl/test/rcl/test_log_level.cpp

```diff
@@ -0,0 +1,207 @@
+  EXPECT_EQ(1ul, log_levels->num_logger_settings);
+  EXPECT_STREQ("rcl", log_levels->logger_settings[0].name);
+  EXPECT_EQ(RCUTILS_LOG_SEVERITY_INFO, log_levels->logger_settings[0].level);
+}
+
+TEST(TestLogLevel, multiple_log_level_with_default_at_front) {
+  rcl_log_levels_t * log_levels = NULL;
+  GET_LOG_LEVEL_FROM_ARGUMENTS(
+    log_levels, "process_name", "--ros-args",
+    "--log-level", "debug", "--log-level", "rcl:=debug");
+  OSRF_TESTING_TOOLS_CPP_SCOPE_EXIT(
+  {
+    EXPECT_EQ(RCL_RET_OK, rcl_log_levels_fini(log_levels));
+  });
+  EXPECT_EQ(RCUTILS_LOG_SEVERITY_DEBUG, log_levels->default_logger_level);
+  EXPECT_EQ(1ul, log_levels->num_logger_settings);
+  EXPECT_STREQ("rcl", log_levels->logger_settings[0].name);
+  EXPECT_EQ(RCUTILS_LOG_SEVERITY_DEBUG, log_levels->logger_settings[0].level);
+}
+
+TEST(TestLogLevel, multiple_log_level_with_default_at_back) {
+  rcl_log_levels_t * log_levels = NULL;
+  GET_LOG_LEVEL_FROM_ARGUMENTS(
+    log_levels, "process_name", "--ros-args",
+    "--log-level", "rcl:=debug", "--log-level", "debug");
+  OSRF_TESTING_TOOLS_CPP_SCOPE_EXIT(
+  {
+    EXPECT_EQ(RCL_RET_OK, rcl_log_levels_fini(log_levels));
+  });
+  EXPECT_EQ(RCUTILS_LOG_SEVERITY_DEBUG, log_levels->default_logger_level);
+  EXPECT_EQ(1ul, log_levels->num_logger_settings);
+  EXPECT_STREQ("rcl", log_levels->logger_settings[0].name);
+  EXPECT_EQ(RCUTILS_LOG_SEVERITY_DEBUG, log_levels->logger_settings[0].level);
+}
+
+TEST(TestLogLevel, multiple_log_level_rightmost_prevail) {
+  rcl_log_levels_t * log_levels = NULL;
+  GET_LOG_LEVEL_FROM_ARGUMENTS(
+    log_levels, "process_name", "--ros-args",
+    "--log-level", "debug", "--log-level", "info",
+    "--log-level", "rcl:=debug", "--log-level", "rcl:=info");
+  OSRF_TESTING_TOOLS_CPP_SCOPE_EXIT(
+  {
+    EXPECT_EQ(RCL_RET_OK, rcl_log_levels_fini(log_levels));
+  });
+  EXPECT_EQ(RCUTILS_LOG_SEVERITY_INFO, log_levels->default_logger_level);
+  EXPECT_EQ(1ul, log_levels->num_logger_settings);
+  EXPECT_STREQ("rcl", log_levels->logger_settings[0].name);
+  EXPECT_EQ(RCUTILS_LOG_SEVERITY_INFO, log_levels->logger_settings[0].level);
+}
+
+TEST(TestLogLevel, log_level_dot_logger_name) {
+  rcl_log_levels_t * log_levels = NULL;
+  GET_LOG_LEVEL_FROM_ARGUMENTS(
+    log_levels, "process_name", "--ros-args",
+    "--log-level", "test.abc:=info");
+  OSRF_TESTING_TOOLS_CPP_SCOPE_EXIT(
+  {
+    EXPECT_EQ(RCL_RET_OK, rcl_log_levels_fini(log_levels));
+  });
+  EXPECT_EQ(RCUTILS_LOG_SEVERITY_UNSET, log_levels->default_logger_level);
+  EXPECT_EQ(1ul, log_levels->num_logger_settings);
+  EXPECT_STREQ("test.abc", log_levels->logger_settings[0].name);
+  EXPECT_EQ(RCUTILS_LOG_SEVERITY_INFO, log_levels->logger_settings[0].level);
+}
+
+int main(int argc, char ** argv)
+{
+  ::testing::InitGoogleTest(&argc, argv);
+  int ret = RUN_ALL_TESTS();
+  // We need to do rcutils_logging_shutdown() since the tests above may
+  // have initialized logging.
+  if (rcutils_logging_shutdown() != RCUTILS_RET_OK) {
```

Comment:
if you need this, please write a TestCase clase with the proper setup and teardown, instead of writing a main

Reply:
I added that.

To be clear, we only want to do this once at the end of the program.  Is there a particular reason to do it in a class rather than in a main?  It seems like a whole lot more boilerplate code to me and the benefit is unclear to me, since we don't need the class otherwise.

Reply:
> To be clear, we only want to do this once at the end of the program. Is there a particular reason to do it in a class rather than in a main? It seems like a whole lot more boilerplate code to me and the benefit is unclear to me, since we don't need the class otherwise.

I didn't see that comment. Both ways are fine, I typically avoid writing a main in `gtest` but there's no big difference.